### PR TITLE
complete the interface of network discover.

### PR DESCRIPTION
**Pull request by Hyzhou**
Signed-off-by: hyzhou.zhy <hyzhou.zhy@alibaba-inc.com>

### CODE REVIEW

File: network/api.go [hunk 1]
```diff
@@ -23,12 +23,10 @@ const (
...
 	joinPath           = "/NetworkDriver.Join"
 	leavePath          = "/NetworkDriver.Leave"
-	//discoverNewPath    = "/NetworkDriver.DiscoverNew"
-	//discoverDeletePath = "/NetworkDriver.DiscoverDelete"
-
+	discoverNewPath    = "/NetworkDriver.DiscoverNew"
+	discoverDeletePath = "/NetworkDriver.DiscoverDelete"
 )
 
-// Driver represent the interface a driver must fulfill.
```
Comment (dave-tucker) [inline network/api.go]: Don't think that comment should be deleted, golint will fail otherwise

File: network/api.go [hunk 2]
```diff
@@ -134,6 +134,18 @@ type LeaveRequest struct {
 	Options    map[string]interface{}
 }
 
+// DiscoverNew is a notification for a new discovery event, Example:a new node joining a cluster
+type DiscoverNewRequest struct {
+	discoverType int
```
Comment (dave-tucker) [inline network/api.go]: DiscoveryType

File: network/api.go [hunk 3]
```diff
@@ -134,6 +134,18 @@ type LeaveRequest struct {
...
+	data         interface{}
```
Comment (dave-tucker) [inline network/api.go]: `DiscoveryData` - and we should have a struct for DiscoveryData too

File: network/api.go [hunk 4]
```diff
@@ -134,6 +134,18 @@ type LeaveRequest struct {
 	Options    map[string]interface{}
 }
 
+// DiscoverNew is a notification for a new discovery event, Example:a new node joining a cluster
+type DiscoverNewRequest struct {
```
Comment (dave-tucker) [inline network/api.go]: We should just make this `DiscoverRequest` as the same struct can be used for both New and Delete

**Comment by dave-tucker** [comment]
Thanks for the contribution @Hyzhou, I've added some comments. Don't worry about the CI failure... it should start working again once you push an update!

**Comment by Hyzhou** [comment]
@dave-tucker Thanks for review, I will fix it soon.

**Comment by Hyzhou** [comment]
It looks some problem with CI. Would you like to help me to review the code. @dave-tucker

**Comment by dave-tucker** [comment]
Hi @Hyzhou - I'm really sorry, but somehow this functionality made it in as a side effect of #19 :sob:
Therefore, I'm going to close this PR. Thanks for your contribution and I hope this doesn't discourage you from contributing in the future!

**Pull request closed without merging**